src/main/java/org/apache/sysds/runtime/transform/encode/MultiColumnEncoder.java

```diff
@@ -314,11 +314,12 @@ public MatrixBlock apply(CacheBlock<?> in) {
 	public MatrixBlock apply(CacheBlock<?> in, int k) {
 		// domain sizes are not updated if called from transformapply
 		boolean hasUDF = _columnEncoders.stream().anyMatch(e -> e.hasEncoder(ColumnEncoderUDF.class));
+		boolean hasWE = _columnEncoders.stream().anyMatch(e -> e.hasEncoder(ColumnEncoderWordEmbedding.class));
 		for(ColumnEncoderComposite columnEncoder : _columnEncoders)
 			columnEncoder.updateAllDCEncoders();
 		int numCols = getNumOutCols();
 		long estNNz = (long) in.getNumRows() * (hasUDF ? numCols : (long) in.getNumColumns());
-		boolean sparse = MatrixBlock.evalSparseFormatInMemory(in.getNumRows(), numCols, estNNz) && !hasUDF;
+		boolean sparse = MatrixBlock.evalSparseFormatInMemory(in.getNumRows(), numCols, estNNz) && !hasUDF && !hasWE;
```

Comment:
actually the "&& !hasWE" is not needed anymore, cuz I am using in the this version "quickSetValue" which should work for both sparse and dense, correct?

Reply:
Yes. For UDF encoders we forced dense as we cannot derive the sparsity of the outputs. But for embeddings, you already know the number of nonzeros.